File: python/valuecell/server/api/schemas/short_cycle_context_event.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class ShortCycleContextEventItemData(BaseModel):
    event_id: int
    user_id: str
    ticker: str
    display_name: str
    topic_name: str | None = None
    layer: str
    event_type: str
    source: str
    occurred_at: datetime
    ingested_at: datetime
    importance_score: int
    direction: str
    time_horizon: str
    tradeability_hint: str | None = None
    summary: str
    payload_json: dict[str, Any] = Field(default_factory=dict)
    record_date: str
    is_active: bool


class ShortCycleContextEventListData(BaseModel):
    generated_at: datetime
    items: list[ShortCycleContextEventItemData] = Field(default_factory=list)
    count: int


class ShortCycleContextEventRefreshRequest(BaseModel):
    ticker: str | None = None

File: frontend/src/types/short-cycle-context-event.ts
export type ShortCycleContextEvent = {
  event_id: number;
  user_id: string;
  ticker: string;
  display_name: string;
  topic_name: string | null;
  layer: string;
  event_type: string;
  source: string;
  occurred_at: string;
  ingested_at: string;
  importance_score: number;
  direction: string;
  time_horizon: string;
  tradeability_hint: string | null;
  summary: string;
  payload_json: Record<string, unknown>;
  record_date: string;
  is_active: boolean;
};

export type ShortCycleContextEventList = {
  generated_at: string;
  items: ShortCycleContextEvent[];
  count: number;
};
